Document Servala cluster setups (#164)
vshn/application-catalog-docs: docs/modules/ROOT/pages/how-tos/add-servala-control-cluster.adoc

<3> The generated steward URL from control.vshn.net

== Vcluster configuration

All the configuration in this chapter has to be done on the vcluster itself.

=== Make sure ArgoCD bootstraps properly

After the vcluster bootstraps, it's very likely that ArgoCD hangs reconfiguring itself.
It's not able to apply some CRDs.
To fix this connect to the vcluster, by port-forwarding the service `servala-control-plane` in the namespace `servala-control-plane`.
A kubeconfig can be found in the same namespace in the secret `vc-servala-control-plane`.

Then clone the rendered cluster manifests from git.vshn.net.
Apply ArgoCD's manifests again with server-side-apply.

.Fix stuck ArgoCD bootstrapping
[source,bash]
----
kubectl apply -Rf manifests/argocd --server-side
----

Connect to the vcluster's ArgoCD instance to check if the sync now runs through.
Afterward all the other apps should also start to sync.

=== Create service cluster kube configs

Create all the kubeconfigs for all the service clusters that will be managed by this control plane.

The followin script assumes, that you use https://git.vshn.net/vshn/openshift4-clusters[OpenShift4 Access].

> You need to connect to each cluster and run the script.

Reviewer: ```suggestion
The following script assumes, that you use https://git.vshn.net/vshn/openshift4-clusters[OpenShift4 Access].
```

> = Add a servala control cluster

Reviewer: ```suggestion
= Add a Servala Control Cluster
```

> Currently service clusters run as vclusters on one of the CSP's service clusters.
> Usually it runs on the first provisioned cluster.

Reviewer: ```suggestion
The Servala Control Cluster (aka control-plane) runs in a vcluster on one of the CSP's service clusters, usually on the first provisioned service cluster.
```

I'd refrain from using "currently" in such documentation. This always implies some uncertainty. Document the fact, plans belong in Jira.

Setting up a new control plane will involve configurations on the service cluster as well as on the vcluster itself.

== Setup the vcluster on the service cluster

This configuration has to be added to the service cluster, in order for a vcluster to be provisioned.

The base service cluster configuration also needs to be applied as well.

=== Register cluster

Register the new cluster on the https://control.vshn.net/syn/lieutenantclusters/vshn-lieutenant-prod[control panel].

> ID: See https://kb.vshn.ch/app-catalog/adr/0031-naming-scheme-for-servala-cluster-names-and-urls.html#_decision[ADR]

Reviewer: ```suggestion
ID: See xref:adr/0031-naming-scheme-for-servala-cluster-names-and-urls.adoc#_decision[ADR]
```

Use the `xref:` scheme as we're interlinking in the same docs =) and in the other places as well of course

> Sales Order: S10783

Reviewer: Somewhere in VSHN Wiki and a link here to there
